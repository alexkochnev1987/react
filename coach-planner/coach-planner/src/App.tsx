import "./App.css";
import { MyRouterProvider } from "./router/router-provider";

function App() {
  return <MyRouterProvider />;
}

export default App;
