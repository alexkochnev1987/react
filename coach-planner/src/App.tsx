import { ThemeProviderComponent } from './theme/ThemeProviderComponent';
import { MyRouterProvider } from './router/router-provider';
import { Provider } from 'react-redux';
import './index.css';
import './firebase';
import { store } from './store/store';
import React from 'react';

function App() {
  return (
    <React.StrictMode>
      <Provider store={store}>
        <ThemeProviderComponent>
          <MyRouterProvider />
        </ThemeProviderComponent>
      </Provider>
    </React.StrictMode>
  );
}

export default App;
