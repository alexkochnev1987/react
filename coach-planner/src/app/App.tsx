import { ThemeProvider } from './providers/ThemeProvider';
import { Provider } from 'react-redux';
import './styles/index.scss';
import '../firebase';
import { store } from '../store/store';
import React from 'react';
import { AppRouter } from './providers/RouterProvider';

function App() {
  return (
    <React.StrictMode>
      <Provider store={store}>
        <ThemeProvider>
          <AppRouter />
        </ThemeProvider>
      </Provider>
    </React.StrictMode>
  );
}

export default App;
