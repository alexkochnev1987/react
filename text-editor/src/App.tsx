import { Layout } from './components/layout/Layout';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import ErrorPage from './pages/Error-page';
import { Login } from './pages/Login';
import { Users } from './components/users/users';
import React, { createContext } from 'react';
import { Provider } from 'react-redux';
import { store } from './store/store';
import MobxStore from './store/mob-x/store';
import { Registration } from './pages/Registration';
// Import the functions you need from the SDKs you need
import { initializeApp } from 'firebase/app';
import { getAnalytics } from 'firebase/analytics';
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: 'AIzaSyC4UE2w0EbmTIeo7mu4k8QsdYDfxehi8t8',
  authDomain: 'coach-planner.firebaseapp.com',
  projectId: 'coach-planner',
  storageBucket: 'coach-planner.appspot.com',
  messagingSenderId: '453833960134',
  appId: '1:453833960134:web:61e09c9f132ce6b43885c7',
  measurementId: 'G-X0BKFHKKD3',
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
const analytics = getAnalytics(app);

interface Store {
  store: MobxStore;
}

const mobxStore = new MobxStore();

export const Context = createContext<Store>({ store: mobxStore });

const router = createBrowserRouter([
  {
    path: '/',
    element: <Layout />,
    errorElement: <ErrorPage />,
    children: [
      {
        path: '/login',
        element: <Login />,
      },
      {
        path: '/registration',
        element: <Registration />,
      },
      {
        path: '/users',
        element: <Users />,
      },
    ],
  },
]);

function App() {
  return (
    <React.StrictMode>
      <Provider store={store}>
        <Context.Provider value={{ store: mobxStore }}>
          <RouterProvider router={router} />
        </Context.Provider>
      </Provider>
    </React.StrictMode>
  );
}

export default App;
