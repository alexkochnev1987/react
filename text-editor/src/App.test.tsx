import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

describe('Routing test', () => {
  it('class btn length', async () => {
    render(<App />);

    const button = screen.getByRole('button');

    expect(button).toBeInTheDocument();
    await userEvent.click(button);
    const navbar = screen.queryByTestId('navbar');
    expect(navbar).toBeInTheDocument();
    const goToUsers = screen.queryByText('Пользователи');
    expect(goToUsers).toBeInTheDocument();
    if (goToUsers) await userEvent.click(goToUsers);

    const users = await screen.findAllByTestId('user-item');
    expect(users.length).toBe(10);

    await userEvent.click(button);
    const goToLogin = await screen.queryByText('Войти');
    expect(goToLogin).toBeInTheDocument();

    if (goToLogin) await userEvent.click(goToLogin);
    const login = await screen.queryByText('Add');

    expect(login).toBeInTheDocument();
  });
});
