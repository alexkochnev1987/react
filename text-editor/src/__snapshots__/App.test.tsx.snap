// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Routing test > class btn length 1`] = `
<input
  placeholder="enter name"
  type="text"
  value=""
/>
`;
